src/document.h

```diff
@@ -525,6 +525,7 @@ struct Document {
                 SetSelect(drawroot->parent->grid->FindCell(drawroot));
         }
         while (len < drawpath.size()) drawpath.remove(0);
+        if (onlysetupdrawpath) return;

```

Comment:
In how many places is `Zoom` called? if only few, it may be cleaner than rather using a boolean argument, to just split this function in 2 (e.g. `ZoomSetDrawPath` and `ZoomDraw`) and call those as needed.

Just an idea.. boolean args to a function that make it take different paths thru the function is usually not so great, often better to split or specialize the function for the different callers instead

Reply:
`Zoom` is called from five places in the code. In every instance,  we have a complete layout calculation and the draw of the selection. So currently it makes sense to split out the setting of the `drawpath` for the initial zooming but not to move the other part of the `Zoom` function into a second separate split function. Do you agree?
Thanks for your feedback in advance!

Reply:
See the updated commit in the PR :-)

Reply:
sure.